execution.py

```diff
@@ -13,6 +13,30 @@
 
 import comfy.model_management
 
+def force_bhw3(image):
+    #convert [CHW, BCHW, CWH] to BHW3
+    was_list = False
+    
+    while isinstance(image, list):
+        was_list = True
+        image = image[0]
+    
+    if len(image.shape) == 3:
+        #add batch dimension
+        image = image.unsqueeze(0)
+    
+    if image.shape[1] == 3:
```

Comment:
What if somebody just happens to have a 3x3 image somewhere? This will corrupt it.

I imagine if check [2] and [3] are both >4 then the conversion is confident. But then the conversion only happens on larger images, and quietly doesn't apply to smaller images.
... This type of thing is a good example of why it might be better to warn than to try to autocorrect - I don't think there's actually a 100% reliable detection of format in all cases, just "99% of the time it's right" heuristic checks like this one.

Reply:
(Also mentioned in Discord, but copying here for documentation)
+1 for just warning (and eventually erroring) in any of the slightly ambiguous cases. If we're going to auto fix anything, it should **just** be the BWH case (which is easy to differentiate due to the tensor only having 3 dimensions). That's probably the most common error anyway.